core/src/main/java/org/radargun/traits/WaitLifecycle.java

```diff
@@ -0,0 +1,19 @@
+package org.radargun.traits;
+
+/**
+ * Lifecycle can be started later if it is not running.
+ *
+ * @author Diego Lovison &lt;dlovison@redhat.com&gt;
+ */
+public abstract class WaitLifecycle implements Lifecycle {
```

Comment:
I prefer because having that class I can see what are the Lifecycle that support later startup.

Reply:
Is there a reason that all `Lifecycle` implementations shouldn't support this?

Reply:
Nops. Looking for `shouldWait(boolean wait)` or `shouldWait()` won't show what are the LIfecycle that support wait.
Looking for `WaitLifecycle` you will see `InfinispanLifecycle`

Reply:
After looking at the code some more, I still just don't having the `WaitLifecycle` class. The default implementation of `shouldWait()` means that any `Lifecycle` implementations that don't care about waiting will behave correctly, and if you need this functionality then implement these methods. Not sure I see a huge benefit to being able to search for this, since it just complicates the class hierarchy of the `InfinispanLifecycle` class.

Reply:
Removed and move the code to `InfinispanLifecycle`